src/figuresClasses.ts

```diff
@@ -1,11 +1,100 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  color: string;
+
+  sideA: number;
+
+  sideB: number;
+
+  sideC: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle sides must be greater than zero.');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} cannot form a triangle.`);
+    }
+
+    this.color = color;
+    this.sideA = a;
+    this.sideB = b;
+    this.sideC = c;
```

Comment:
In TypeScript, when you declare a constructor parameter with an access modifier, it implicitly creates a property on the class with the same name. There is no need to assign the parameter values to class properties manually. You can simplify the constructor by using access modifiers in the parameters.